src/common/lib/client/rest.ts

```diff
@@ -228,6 +230,155 @@ class Rest {
     }
   }
 
+  batchPublish(
+    specs: API.Types.BatchPublishSpec,
+    callback: API.Types.StandardCallback<
+      API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>
+    >
+  ): void;
+  batchPublish(
+    specs: API.Types.BatchPublishSpec[],
+    callback: API.Types.StandardCallback<
+      API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]
+    >
+  ): void;
+  batchPublish(
+    specs: API.Types.BatchPublishSpec
+  ): Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>>;
+  batchPublish(
+    specs: API.Types.BatchPublishSpec[]
+  ): Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]>;
+  batchPublish(
+    specOrSpecs: API.Types.BatchPublishSpec | API.Types.BatchPublishSpec[],
+    callbackArg?:
+      | API.Types.StandardCallback<
+          API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>
+        >
+      | API.Types.StandardCallback<
+          API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]
+        >
+  ):
+    | void
+    | Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>>
+    | Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]> {
+    if (callbackArg === undefined) {
+      if (this.options.promises) {
+        return Utils.promisify(this, 'batchPublish', [specOrSpecs]) as
+          | Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>>
+          | Promise<API.Types.BatchResult<API.Types.BatchPublishSuccessResult | API.Types.BatchPublishFailureResult>[]>;
+      }
+      callbackArg = noop;
+    }
+
+    const callback = callbackArg;
+
+    let requestBodyDTO: API.Types.BatchPublishSpec[];
+    let singleSpecMode: boolean;
+    if (Array.isArray(specOrSpecs)) {
```

Comment:
we use a more portable version of this for compatibility
```suggestion
    if (Utils.isArray(specOrSpecs)) {
```

Reply:
Cool, have updated.